### ISSUE TITLE
`webpack --watch` builds only once and exits immediately without even any warnings&errors.

### ISSUE THREAD

**Issue by wonderbeyond** [webpack/webpack#2297]
`webpack --watch` builds only once and then exits immediately.
However, It should keeps running and listening to fs changes for recompilation as expected.

I cannot see any errors or warnings even if I enable the verbose mode by:

``` shell
webpack --watch --display-error-details --display-reasons
```

Below is my environment infomation:

``` shell
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.4 LTS
Release:    14.04
Codename:   trusty

$ uname -a
Linux ubuntu 3.19.0-56-generic #62~14.04.1-Ubuntu SMP Fri Mar 11 11:03:15 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

$ node -v
v4.4.1

$ npm -v
2.14.20

$ which node
~/.nvm/versions/node/v4.4.1/bin/node

$ which webpack  # Version: webpack 1.12.14
./node_modules/.bin/webpack

$ npm ls --depth=0
├── css-loader@0.23.1
├── exports-loader@0.6.3
├── file-loader@0.8.5
├── handlebars@4.0.5
├── handlebars-loader@1.2.0
├── handlebars-template-loader@0.5.7
├── image-webpack-loader@1.6.3
├── iscroll@5.2.0
├── node-sass@3.4.2
├── normalize.css@4.0.0
├── raw-loader@0.5.1
├── sass-loader@3.2.0
├── style-loader@0.13.1
├── vue@1.0.20
├── webpack@1.12.14
├── zepto@1.0.1
└── zepto-modules@0.0.0
```

**Comment by artburkart** [comment]
This sounds like an `inotify` file watch limit issue? Here is a helpful [wiki article](https://github.com/guard/listen/wiki/Increasing-the-amount-of-inotify-watchers) from @piotrzaborow that may help.

Perhaps `webpack` shouldn't fail silently when it runs out of watches?

**Comment by wonderbeyond** [comment]
@artburkart Many thanks! Setting `fs.inotify.max_user_watches` solves my problem.

> Perhaps webpack shouldn't fail silently when it runs out of watches?

Yes, I agree!

**Comment by wangxiexe** [comment]
I am running on ubuntu on windows 10

wangxi@WANGXI-SURFACEB:/proc/sys/kernel$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.4 LTS
Release:        14.04
Codename:       trusty

and seems I can not set max watchers

wangxi@WANGXI-SURFACEB:~/test$ sudo sysctl -p
sudo: unable to resolve host WANGXI-SURFACEB
sysctl: cannot stat /proc/sys/fs/inotify/max_user_watches: No such file or directory
sysctl: cannot stat /proc/sys/fs/inotify/max_user_watches: No such file or directory
sysctl: cannot stat /proc/sys/fs/inotify/max_user_watches: No such file or directory

is there any way I can workaround this ?

**Comment by chipit24** [comment]
I am also running the new [Ubuntu on Windows](https://msdn.microsoft.com/en-us/commandline/wsl/about) and webpack runs once and exits immediately even with the `--watch` flag. The solution suggested by @artburkart doesn't work in this situation.

**Comment by adammarcwilliams** [comment]
I am running the new Ubuntu on Windows too and get the same problem with --watch.

I've also ran the webpack-dev-server which work fine but will not watch for changes either.

**Comment by lpil** [comment]
I've also run into this problem. I think this situation should be considered an error and displayed to the user.

**Comment by gilamran** [comment]
Same here, running latest Ubuntu support on windows 10. no watch... no error

**Comment by stevenleeg** [comment]
I'm seeing this same behavior on Ubuntu.

```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.1 LTS
Release:        16.04
Codename:       xenial

$ node -v
v6.7.0

$ npm ls webpack
lousyui@1.0.0 /home/steve/Code/lousyui
└── webpack@1.13.2 
```

Any ideas as to what could be going on here? I don't see any errors/warnings when running webpack and it appears to compile just fine on the first (and only) build.

**Comment by jaafit** [comment]
Had the same issue in Ubuntu.  

For me it worked if I added the -d option: "webpack -d --watch"

**Comment by orazdow** [comment]
I'm running just in windows 10 cmd. I have the same problem, watch mode just compiles and exits...

**Comment by ratiotile** [comment]
I got the same issue in Debian 8. The inotify watch limit default is too low. 

Webpack --watch should give a warning and not just fail silently.

**Comment by qertis** [comment]
I got the same issue in Windows 10. 

```[02:02:15] Version: webpack 2.3.1
                  Asset     Size  Chunks             Chunk Names
5e908b173d38a83d1cb0.js  2.65 kB       0  [emitted]  main
[02:02:15] webpack is watching for changes
qertis@FUCKING-WIN10:/mnt/c/Users/iavto/Documents/Works/polymer-starter-kit$ 
```

**Comment by J4K0B** [comment]
> Try this workaround:
>
> Add this to your webpack configuration file:
>
>    watchOptions: {
>        poll: true
>    },

Works for me on Ubuntu on Windows.

[Lahdo](https://github.com/lahdo) posted that [here](https://github.com/webpack/webpack/issues/2949)

**Comment by chipit24** [comment]
See the answer by Rich Turner on SO here: http://stackoverflow.com/questions/38922582/running-webpack-with-watch-in-bash-on-ubuntu-on-windows-builds-once-and-exits

> Looks like you're running Windows 10 Anniversary Update which didn't yet support INOTIFY file-change event notifications. This capability was added in Insider build 14942 back in Oct 2016:
>
> https://blogs.msdn.microsoft.com/commandline/2016/10/07/wsl-adds-inotify-filesystem-change-notification-support/
>
> Once you upgrade to a recent Insider build, or to Creators Update which will be released this spring, you'll be able to enjoy INOTIFY support which is more efficient than the polling mechanism that your config change enabled.
>
> HTH.

**Comment by errorx666** [comment]
I have this issue on Windows 10 but I'm not using the Linux subsystem.  I'm using PowerShell.

**Comment by lawrenceagles** [comment]
Experiencing this problem with antergos Linux.

**Comment by lawrenceagles** [comment]
@artburkart  your suggestion worked for me.

**Comment by jdavidbakr** [comment]
Experiencing this on Ubuntu Budgie, but @jaafit your solution worked for me (adding `-d` flag to the script)

**Comment by nicolasxu-bg** [comment]
I am having the same problem 

"webpack --watch --progress --hot --mode development "

**Comment by tzatter** [comment]
it works for me to add this flag "--watch-poll"
"webpack --watch-poll --watch"

reference for --watch-poll
[https://webpack.js.org/configuration/watch/#watchoptions-poll](https://webpack.js.org/configuration/watch/#watchoptions-poll)

> If watching does not work for you, try out this option. Watching does not work with NFS and machines in VirtualBox.

**Comment by Perlover** [comment]
Hi,

I noticed same problem
But after researching i found that it's not bug
If filesystem is mounted with `noatime` option (this option is recommended in many sources for SSD disks as optimization) this plugin deoesn't work as `webpack --watch` (only `webpack --watch-poll --watch`)
So i recommend to check /etc/fstab file in Linux (Unix) and remove `noatime` if you have and reboot

**Comment by MartinElvar** [comment]
Same issue with Pop! OS.

@Perlover Is right on, but the polling solution does come at a resource expense.

**Comment by amontalenti** [comment]
Why is this issue closed? This happened to me on Ubuntu 18.04 using webpack 4.19.0. This was a completely maddening issue that cost me a solid hour of banging my head against the wall, because webpack exited 0 with no debug or error information. In my case, it only exited when I used it under `concurrently` when running with another (Python) development server that also used inotify watches.

The solution, for me, was to run this:

```
$ echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p
```

But... the root issue here isn't my system; it is, instead, that webpack is not *reporting* that it has run out of inotify watches; instead, it is just silently exiting.

**Comment by amontalenti** [comment]
Just noticed "If you think this is still a valid issue, please file a new issue with additional information." by the webpack bot, so opened a new issue, #8101.
